src/compat.1.8.jl

```diff
@@ -0,0 +1,5 @@
+# https://github.com/JuliaLang/julia/pull/25085
```

Comment:
These are needed to pass tests only for v1.8+, but it should be ok to use also for all other versions, right?  If so, can probably go outside of the conditional.  Apart from that, looks good.  Thanks!